Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Base class Interactable GameObjects can inherit from.
+    /// </summary>
+    public class Interactable : MonoBehaviour
+    {
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized += KeywordRecognized;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized -= KeywordRecognized;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any gameObject.
+        /// </summary>
+        /// <param name="go">GameObject user has tapped.</param>
+        protected virtual void OnTap(GameObject go)
+        {
+            if (go == gameObject)
+            {
+                // Do something if our gameObject has been tapped
+            }
+            else
+            {
+                // Do something if we've tapped any object except for our own
+            }
+        }
+
+        /// <summary>
+        /// Called when a user's gaze enters any gameObject.
+        /// </summary>
+        /// <param name="go">The GameObject that the users gaze has entered.</param>
+        protected virtual void OnGazeEnter(GameObject go)
+        {
+            if (go == gameObject)
+            {
+                // Do something if our gaze has entered this gameObject
+            }
+            else
+            {
+                // Do something if our gaze has entered another gameObject
+            }
+        }
+
+        /// <summary>
+        /// Called when a user's gaze exits any gameObject.
+        /// </summary>
+        /// <param name="go">The GameObject that the users gaze has left.</param>
+        protected virtual void OnGazeExit(GameObject go)
+        {
+            if(go == gameObject)
+            {
+                // Do something if our gaze has left this gameObject
+            }
+            else
+            {
+                // Do something if our gaze has left another gameObject
+            }
+        }
+
+        /// <summary>
+        /// Called when the Keyword Manager recognizes a spoken keyword.
+        /// </summary>
+        /// <param name="keyword">Spoke Keyword.</param>
+        protected virtual void KeywordRecognized(string keyword)
+        {
+            if (keyword.Equals("string"))
```

Comment:
I still think this function should be empty, maybe just have a "do nothing" comment or an explanation.
If I don't override this method, I don't expect the base class to do anything.